std/algorithm/iteration.d

```diff
@@ -902,67 +913,100 @@ template each(alias pred = "a")
             size_t i = 0;
             while (!r.empty)
             {
-                cast(void) binaryFun!BinaryArgs(i, r.front);
+                static if (!is(typeof(binaryFun!BinaryArgs(i, r.front)) ==
+                        Flag!"each"))
+                    cast(void) binaryFun!BinaryArgs(i, r.front);
```

Comment:
It's a common stylistic practice in other languages, in some it is required. The function shouldn't be returning a significant value anyway.